### PR TITLE
Pre-commit lint check

### PR DESCRIPTION
**Pull request by DurkoMatko**
**add**: pre-commit "hook" to run linter on staged files

Rather than putting the script to _.git/hooks/pre-commit.sh_ I'm using npm package for this. According to this thread (https://stackoverflow.com/questions/3462955/putting-git-hooks-into-repository), git hooks shouldn't be a part of the repository and I wanted to avoid creating additional step in the dev setup. (e.g copying the pre-commit script to the _.git/hooks/_ or creating symlinks etc). This way, contributors don't have to do anything. Only staged files get checked

### CODE REVIEW

**Comment by mcgnly** [comment]
I have also done this with husky, which lets it all run like yarn scripts, without the need for a separate .sh. How we're doing it in my work project is like this:


```{
  "dependencies": {
 ...
  },
  "scripts": {
    "build:tailwind": "postcss src/styles/index.scss -o src/styles/index.generated.scss",
    "watch:tailwind": "postcss -w src/styles/index.scss -o src/styles/index.generated.scss",
    "start": "run-p watch:tailwind start:react",
    "start:react": "react-scripts start",
    "prebuild": "env NODE_ENV=production yarn build:tailwind",
    "build": "react-scripts build",
    "build:style": "tailwind build ",
    "test:unit": "react-scripts test",
    "eject": "react-scripts eject",
    "format:check": "prettier --check \"src/**/*.js\"",
    "format:write": "prettier --write \"src/**/*.js\""
  },
  "eslintConfig": {
    "extends": "react-app"
  },
  "browserslist": {
    "production": [
...
    ],
    "development": [
...
    ]
  },
  "devDependencies": {
    "husky": "^4.2.5",
    "lint-staged": "^10.2.10",
    "npm-run-all": "^4.1.5",
    "postcss-import": "^12.0.1",
    "postcss-scss": "^2.1.1",
    "prettier": "2.0.5",
    "tailwindcss": "^1.4.6"
  },
  "husky": {
    "hooks": {
      "pre-commit": "lint-staged"
    }
  },
  "lint-staged": {
    "src/**/*.{js,json,scss,md}": [
      "yarn run format:write"
    ]
  }
}
```

**Comment by DurkoMatko** [comment]
@mcgnly Ok I've changed this to use the husky, thx for the tip!..also, I've let prettier to format our files according to the current config.

File: .eslintrc [hunk 1]
```diff
@@ -23,6 +23,7 @@
     "react/prop-types": "off",
     // should be turned on eventually, but not compatible with current codebase
     "no-shadow": "off",
+    "no-console":"off",
```
Comment (mcgnly) [inline .eslintrc]: I'm fine with this being a warn but not an error? but we can also leave it off for now

File: .prettierrc [hunk 2]
```diff
@@ -1,5 +1,6 @@
 {
   "printWidth": 100,
   "singleQuote": false,
-  "doubleQuote": true
+  "doubleQuote": true,
+  "semi":false
```
Comment (mcgnly) [inline .prettierrc]: why the no-semicolon rule? I prefer them ¯\_(ツ)_/¯

File: package.json [hunk 3]
```diff
@@ -68,5 +70,15 @@
     "lint-staged": "^10.2.2",
     "prettier": "^2.0.5",
     "typescript": "^3.9.5"
+  },
+  "husky": {
```
Comment (mcgnly) [inline package.json:75]: love it 🎉

**Review by mcgnly — APPROVED** [review]
I personally disagree with the no-semi rule, but I really don't care enough to block a PR over it
Otherwise, looks great! Thanks for doing this :)

File: .prettierrc (hunk 2 above)
Comment (DurkoMatko) [reply .prettierrc]: I prefer them tooo :D I had the impression u don't so I wanted to adjust it the way u feel comfy as you're the main contributor...ok will change it and rerun `prettier --write`

Comment (DurkoMatko) [reply .prettierrc]: Done

File: .eslintrc (hunk 1 above)
Comment (DurkoMatko) [reply .eslintrc]: Changed to warn

**Pull request merged**